db/migrate/20150508140514_create_build_artifact.rb

```diff
@@ -0,0 +1,12 @@
+class CreateBuildArtifact < ActiveRecord::Migration
+  def change
+    create_table :builds do |t|
+      t.belongs_to :project,    null: false, index: true
+      t.string :git_sha,        index: true
+      t.string :git_ref
+      t.string :container_sha,  index: true
+      t.string :container_ref
+      t.timestamps
```

Comment:
Yeah, `[project, git_sha]` should probably be unique. Technically, `git_sha` could probably be unique, if we assume we'll never have 2 projects that point to the same repo.